### PR TITLE
Final coverage fixes

### PR DESCRIPTION
**Pull request by BenFordTytherington**

### CODE REVIEW

File: crates/cxx-qt-gen/src/generator/rust/signals.rs [hunk 1]
```diff
@@ -76,8 +76,7 @@ pub fn generate_rust_signal(
     let self_type_cxx = if signal.mutable {
         parse_quote! { Pin<&mut #qobject_name_rust> }
     } else {
-        // Signals cannot be immutable so this cannot be reached
-        parse_quote! { &#qobject_name_rust }
+        unreachable!("Signals cannot be immutable right now so this cannot be reached")
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/signals.rs:79]: add your start/stop codecov here :-)

File: crates/cxx-qt-gen/src/naming/type_names.rs [hunk 2]
```diff
@@ -801,6 +801,29 @@ mod tests {
...
+        let items = [parse_quote! {
+            mod ffi {
+             enum MyEnum {
+                Yes,
+                No
+            }
+            extern "C++" {
+                type MyEnum;
+            }
+            }
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/naming/type_names.rs:815]: try to format things like Rust would

File: crates/cxx-qt-gen/src/naming/type_names.rs [hunk 3]
```diff
@@ -801,6 +801,29 @@ mod tests {
...
+        }];
+        let mut types = TypeNames::default();
+        // assert!(types
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/naming/type_names.rs:818]: assume this test isn't finished yet :-)

File: crates/cxx-qt-gen/src/parser/mod.rs [hunk 4]
```diff
@@ -102,7 +103,12 @@ impl Parser {
         }
 
         // Create a new module using only items that are not CXX-Qt items
-        module.content = Some((Brace::default(), others));
+        if !others.is_empty() {
+            module.content = Some((Brace::default(), others));
+        } else {
+            module.content = None;
+            module.semi = Some(Semi::default());
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/parser/mod.rs:110]: i think in `writer/rust/mod.rs` where we do `cxx_mod.content = Some(...)` when the `cxx_mod.content` is None. We should then do `cxx_mod.semi = None` to ensure we output the right thing?

File: crates/cxx-qt-gen/src/parser/mod.rs [hunk 5]
```diff
@@ -21,6 +21,7 @@ use crate::{
     syntax::{attribute::attribute_take_path, expr::expr_to_string},
 };
 use cxxqtdata::ParsedCxxQtData;
+use syn::token::Semi;
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/parser/mod.rs:24]: could combine into the use below with `token::{Brace, Semi}`

File: crates/cxx-qt-gen/src/naming/type_names.rs (hunk 3 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/naming/type_names.rs:818]: I think it works, just wanted to be sure. removed the commented code

File: crates/cxx-qt-gen/src/generator/rust/constructor.rs [hunk 6]
```diff
@@ -278,7 +278,13 @@ pub fn generate(
             .map(|mut parameter| -> Result<_> {
                 if let FnArg::Typed(pat_type) = &mut parameter {
                     *pat_type.ty = syn_type_cxx_bridge_to_qualified(&pat_type.ty, type_names)?;
+                } else {
+                    // CODECOV_EXCLUDE_START
+                    // Unreachable as route_arguments_parameters are created with names like arg0: Type above
+                    unreachable!("Args here should not be able to be Receiver")
+                    // CODECOV_EXCLUDE_STOP
                 }
+                // Should this be an error if FnArg::Receiver is passed?
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/constructor.rs:287]: this comment can be removed? there is unreachable now above?

Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/generator/rust/constructor.rs]: yea the args it is getting are built in the block above and get given idents like arg0, arg1, so they could never be self: T or whatever

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/rust/constructor.rs]: cool

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**